pkg/artifactory/resource_artifactory_ldap_setting.go

```diff
@@ -0,0 +1,286 @@
+			},
+			"manager_dn": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "",
+				Description: "The full DN of the user that binds to the LDAP server to perform user searches. Only used with \"search\" authentication.",
+			},
+			"manager_password": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The password of the user that binds to the LDAP server to perform the search. Only used with \"search\" authentication.",
+				Sensitive:   true,
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func resourceLdapSettingsRead(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	rClient := m.(*resty.Client)
+	ldapConfigs := &xmlLdapConfig{}
+	rsrcLdapsetting := unpackLdapSetting(d)
+
+	resp, err := rClient.R().Get("artifactory/api/system/configuration")
+	if err != nil {
+		return diag.Errorf("failed to retrieve data from <base_url>/artifactory/api/system/configuration during Read")
+	}
+
+	err = xml.Unmarshal(resp.Body(), &ldapConfigs)
+	if err != nil {
+		return diag.Errorf("failed to xml unmarshal ldap setting during read operation")
+	}
+	matchedLdapSetting := LdapSetting{}
+	for _, iterLdapSetting := range ldapConfigs.LdapSettings.LdapSettingArr {
+		if iterLdapSetting.Key == rsrcLdapsetting.Key {
+			matchedLdapSetting = iterLdapSetting
+		}
+	}
+
+	packDiag := packLdapSetting(&matchedLdapSetting, d)
+	if packDiag != nil {
+		return packDiag
+	}
+	return diag.Diagnostics{{
+		Severity: diag.Warning,
+		Summary:  "Usage of Undocumented Artifactory API Endpoints",
+		Detail:   "The artifactory_ldap_setting resource uses endpoints that are undocumented and may not work with SaaS environments, or may change without notice.",
+	}}
+}
+
+func resourceLdapSettingsUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	unpackedLdapSetting := unpackLdapSetting(d)
+
+	var constructBody = map[string]map[string]map[string]LdapSetting{}
+	constructBody["security"] = map[string]map[string]LdapSetting{}
+	constructBody["security"]["ldapSettings"] = map[string]LdapSetting{}
+	constructBody["security"]["ldapSettings"][unpackedLdapSetting.Key] = *unpackedLdapSetting
+	content, err := yaml.Marshal(&constructBody)
+
+	if err != nil {
+		return diag.Errorf("failed to marshal ldap settings during Update")
+	}
+
+	err = sendConfigurationPatch(content, m)
+	if err != nil {
+		return diag.Errorf("failed to send PATCH request to Artifactory during Update")
+	}
+
+	// we should only have one ldap setting resource, using same id
+	d.SetId(unpackedLdapSetting.Key)
+	return resourceLdapSettingsRead(ctx, d, m)
+}
+
+func resourceLdapSettingsDelete(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	rClient := m.(*resty.Client)
+	ldapConfigs := &xmlLdapConfig{}
+
+	rsrcLdapSetting := unpackLdapSetting(d)
+
+	response, err := rClient.R().Get("artifactory/api/system/configuration")
+	if err != nil {
```

Comment:
We should also check for response error, e.g. `response.IsError()` (if we don't care which error status code).